### PR TITLE
cpu/stm32: simplify stmclk_disable_hsi function

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

There is no need to check for CLOCK_HSE or to check if HSI is used as SYSCLK, this is already checked at compile time in the clock initialization code of each STM32 families:

https://github.com/RIOT-OS/RIOT/blob/94fec19f8d762d87a6661dfc115b626e5e66754b/cpu/stm32/stmclk/stmclk_f0f1f3.c#L120-L130

https://github.com/RIOT-OS/RIOT/blob/94fec19f8d762d87a6661dfc115b626e5e66754b/cpu/stm32/stmclk/stmclk_f0f1f3.c#L197-L200

This PR is saving 16 bytes on nucleo-l412kb.

### Testing procedure

- A green Murdock
- Check that boards of each affected family groups (f0f1f3, f2f4f7, l0l1, l4wb, g0g4) are still working and preferably boards for which the PLL is clocked by HSI. I would suggest: nucleo-f042k6, nucleo-l073rz, nucleo-l433rc, nucleo-g070rb. I couldn't find any f2f4f7 board not providing HSE.

### Issues/PRs references

Tick an item in #14975

### CODE REVIEW

File: cpu/stm32/stmclk/stmclk_common.c [hunk 1]
```diff
@@ -49,9 +49,6 @@
 #define RCC_CSR_LSIRDY          RCC_CSR_LSI1RDY
 #endif
 
-#ifndef CLOCK_HSE
-#define CLOCK_HSE   (0U)
-#endif
```
Comment (fjmolinas) [inline cpu/stm32/stmclk/stmclk_common.c:54]: Git grep still shows a bunch of occurrences of this define, are you sure it is unused?

**Comment by fjmolinas** [comment]
Removing the extra check makes sense.

File: cpu/stm32/stmclk/stmclk_common.c (hunk 1 above)
Comment (aabadie) [reply cpu/stm32/stmclk/stmclk_common.c:54]: The define is still used but doesn't have the same semantic. CLOCK_HSE contains the frequency of the high speed external and is always defined in the clock configurations (even if there's no HSE).
The existence of the HSE is handled by `CONFIG_BOARD_HAS_HSE` now.

`CLOCK_HSE is now unused in stmclk_common.c` that's why it's removed from there.

Comment (aabadie) [reply cpu/stm32/stmclk/stmclk_common.c:54]: There are also remaining of CLOCK_HSE in clk_conf and doc.txt but those will be handled in separate PRs (same as CLOCK_LSE).

Comment (fjmolinas) [reply cpu/stm32/stmclk/stmclk_common.c:54]: Ok, if it was used somewhere I guess murdock would flag it.

**Review by fjmolinas — APPROVED** [review]
ACK.

**Pull request merged**